packages/core/src/extensions/Blocks/nodes/Block.ts

```diff
@@ -162,275 +109,408 @@ export const Block = Node.create<IBlock>({
+        ({ state, chain }) => {
+          const blockInfo = getBlockInfoFromPos(state.doc, posInBlock);
+          if (blockInfo === undefined) {
+            return false;
+          }
+
+          const { node, startPos, endPos } = blockInfo;
+
+          if (node.textContent.length === 0) {
+            const oldBlockContentPos = startPos + 1;
+
+            return chain()
+              .BNSetContentType(posInBlock, type, attributes)
+              .setTextSelection(oldBlockContentPos)
+              .run();
+          } else {
+            const newBlockInsertionPos = endPos + 1;
+            const newBlockContentPos = newBlockInsertionPos + 1;
+
+            return chain()
+              .BNCreateBlock(newBlockInsertionPos)
+              .BNSetContentType(newBlockContentPos, type, attributes)
+              .setTextSelection(newBlockContentPos)
+              .run();
           }
-          return false;
         },
-      joinBackward:
-        () =>
-        ({ view, dispatch, state }) =>
-          joinBackward(state, dispatch, view), // Override default joinBackward with edited command
     };
   },
+
   addProseMirrorPlugins() {
-    return [PreviousBlockTypePlugin(), OrderedListPlugin()];
+    return [PreviousBlockTypePlugin()];
   },
+
   addKeyboardShortcuts() {
     // handleBackspace is partially adapted from https://github.com/ueberdosis/tiptap/blob/ed56337470efb4fd277128ab7ef792b37cfae992/packages/core/src/extensions/keymap.ts
     const handleBackspace = () =>
       this.editor.commands.first(({ commands }) => [
-        // Maybe the user wants to undo an auto formatting input rule (e.g.: - or #, and then hit backspace) (source: tiptap)
+        // Deletes the selection if it's not empty.
+        () => commands.deleteSelection(),
+        // Undoes an input rule if one was triggered in the last editor state change.
         () => commands.undoInputRule(),
-        // maybe convert first text block node to default node (source: tiptap)
+        // Removes a level of nesting if the block is indented and the block is empty.
         () =>
-          commands.command(({ tr }) => {
-            const { selection, doc } = tr;
-            const { empty, $anchor } = selection;
-            const { pos, parent } = $anchor;
-            const isAtStart = Selection.atStart(doc).from === pos;
-
-            if (
-              !empty ||
-              !isAtStart ||
-              !parent.type.isTextblock ||
-              parent.textContent.length
-            ) {
-              return false;
+          commands.command(({ state }) => {
+            const { depth } = getBlockInfoFromPos(
+              state.doc,
+              state.selection.from
+            )!;
+
+            const selectionAtBlockStart =
+              state.selection.$anchor.parentOffset === 0;
+            const selectionEmpty =
+              state.selection.anchor === state.selection.head;
+            const blockIndented = depth > 2;
```

Comment:
Do we need this check? My guess is `liftListItem` would return false if it can't lift the current block

Reply:
That's a fair point, I added quite a few of these checks/conditions to the keyboard commands since it was easier for me to follow along with what's happening and catch bugs, though it means some of them are redundant. I do still think it helps with readability and the performance hit is negligible, but I'll remove them if you'd rather have less code.

Reply:
Let's comment them out for now, to make sure it doesn't conflict with anything. If it's ok we can remove it later.